copy/copy.go

```diff
@@ -106,9 +113,15 @@ func Image(policyContext *signature.PolicyContext, destRef, srcRef types.ImageRe
 
 	dest, err := destRef.NewImageDestination(options.DestinationCtx)
 	if err != nil {
-		return errors.Wrapf(err, "Error initializing destination %s", transports.ImageName(destRef))
+		retErr = errors.Wrapf(err, "Error initializing destination %s", transports.ImageName(destRef))
+		return
 	}
-	defer dest.Close()
+	defer func() {
+		if err := dest.Close(); err != nil {
+			retErr = errors.Wrapf(retErr, "dest: %v", err)
```

Comment:
… something like this would be nice _if we can make that human-readable_. See how the other `errors.Wrapf` calls in here try to tell the user what is going on. I am not sure that this is workable here:
```
errors.Wrapf(retErr, "Error closing destination: %v, after primary failure", err)
```
is bordering on incomprehensible, especially if `err` were long / created by several `errors.Wrapf` calls. (We can make the underlying data type situation better by using something like `aggregateErr` from `openshift-copies.go`, but that is probably even worse for readability.)

If we can’t make that human-readable, I think preserving a non-`nil` `retErr` is more important than the failure from `.Close`.

@runcom ?